src/main/scala/is/hail/table/Table.scala

```diff
@@ -436,19 +411,18 @@ class Table(val hc: HailContext, val ir: TableIR) {
     copy2(globalSignature = finalSignature, globals = newGlobal)
   }
 
-  def annotate(cond: String): Table = {
-
+  def annotate(code: String): Table = {
     val ec = rowEvalContext()
 
-    val (paths, asts) = Parser.parseAnnotationExprsToAST(cond, ec, None)
+    val (paths, asts) = Parser.parseAnnotationExprsToAST(code, ec).unzip
     if (paths.length == 0)
       return this
 
     val irs = asts.flatMap { x => x.typecheck(ec); x.toIR() }
 
-    if (irs.length != asts.length) {
+    if (irs.length != asts.length && globalSignature.size == 0) {
```

Comment:
the current IR is going to be horribly inefficient if there are large globals (the addAnnotation for each row). This is for performance in the short term, since I expect we have an 0.2 beta before we fix that.

Reply:
don't you want `(irs.length != asts.length || globalSignature.size != 0)` in that case, then?

Reply:
poke on this one? if toIR doesn't work on all the ASTs but you have non-empty globals, then you end up taking the IR path on an incomplete set of IRs (which doesn't seem like what you want?)

Reply:
oh, crap. missed this.